### ISSUE TITLE
Getting DL3045 warning about missing WORKDIR in "child" image but parent has one set

### ISSUE THREAD

**Issue by C-Duv** [hadolint/hadolint#930]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

I am getting a "DL3045 warning: `COPY` to a relative destination without `WORKDIR` set." in a *Dockerfile* using a parent (`FROM`) image that have a `WORKDIR` set.

I am not really sure it can be fixed but I wanted to open an issue for reference.

### Expected behavior

No *DL3045 warning* because child will use/inherit parent's `WORKDIR`.

### Actual behavior

Getting a *DL3045 warning* because the child *Dockerfile* is using relative path.

### Steps to reproduce the behavior

Here are simple shell commands to create both Dockerfile and trigger the warning.

```Shell
IMAGE_NAME_PREFIX="hadolint_dl3045bug"

echo "bar" > ./foo.txt

cat > parent.Dockerfile <<EOF
FROM alpine

WORKDIR /tmp
EOF

cat > child.Dockerfile <<EOF
FROM ${IMAGE_NAME_PREFIX}_parent

COPY foo.txt ./foo.txt
EOF

hadolint --ignore DL3006 parent.Dockerfile child.Dockerfile
```

Output: 

```
child.Dockerfile:3 DL3045 warning: `COPY` to a relative destination without `WORKDIR` set.
```

Here is a simple shell script to build both images an run the child the check it's working just fine: 

```Shell
docker build -t ${IMAGE_NAME_PREFIX}_parent -f parent.Dockerfile .
docker build -t ${IMAGE_NAME_PREFIX}_child -f child.Dockerfile .
docker run --rm -i -t ${IMAGE_NAME_PREFIX}_child /bin/sh -c 'pwd && cat ./foo.txt'
/tmp
bar
```

**Output of `hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.0
```

**Additional environment details (OS, stack version, etc.)**

* OS: Ubuntu 22.04.1 LTS x86_64
* Docker version: 20.10.23, build 7155243

**Comment by m-ildefons** [comment]
Hi Claude,

thanks for the bug report. This is a limitation of the way Hadolint works. Since it only analyzes the Dockerfile at hand and doesn't do image introspection (neither the resulting image, nor the parent image) it can not know that `${IMAGE_NAME_PREFIX}_parent` already defined a working directory.

You can possibly join the two Dockerfiles, in which case Hadolint won't display the warning as it will pick up the earlier `WORKDIR` expression.

```Dockerfile
FROM opensuse/tumbleweed as stage1

WORKDIR /tmp

FROM stage1

COPY foo.txt ./foo.txt
```

If that isn't possible, you can try to avoid using a relative path in the `COPY` statement.

IMO the warning is warranted in the reproducer anyways, since having the code split into two different dockerfiles means it's not entirely obvious that the child dockerfile depends on that `WORKDIR` statement in the parent dockerfile, making this contraption prone to breakage, should the base image of the child dockefile be changed in the future.

Hope this helps.

**Comment by eero-t** [comment]
> You can possibly join the two Dockerfiles, in which case Hadolint won't display the warning as it will pick up the earlier `WORKDIR` expression.

In which Hadolint version this was fixed?   At least it does not work for v2.8 (in Fedora 36)...
